Accessing data from "Nutrition" is not possible for anything except "Carbohydrates"

**Describe the bug**
Requesting "read" permissions for any dietary data but "Carbohydrates" doesn't work.

**To Reproduce**
Steps to reproduce the behavior:

```
const permissions = {
  permissions: {
    read: [
      AppleHealthKit.Constants.Permissions.ActiveEnergyBurned, // THIS IS OK
      AppleHealthKit.Constants.Permissions.BasalEnergyBurned, // THIS IS OK
      AppleHealthKit.Constants.Permissions.EnergyConsumed, 
      AppleHealthKit.Constants.Permissions.Protein,
      AppleHealthKit.Constants.Permissions.BodyMass, // THIS IS OK
      AppleHealthKit.Constants.Permissions.BodyFatPercentage, // THIS IS OK
    ],
    write: [],
  },
};
```

**Expected behavior**
After doing: 

```
AppleHealthKit.initHealthKit(permissions, (error: string) => {
  if (error) {
    console.log(error);
    console.log('[ERROR] Cannot grant permissions!');
  }
});
```

To be presented with a screen that includes the permissions requested (and described in the `permissions` array).


**Screenshots**
Here's a screenshot of code & emulator:
![screenshot](https://i.imgur.com/LhX6oye.png)

**Smartphone (please complete the following information):**

- Device: Simulator:  iphone 12 mini
- OS: 14.4
- Version idk

**Additional context**
I copy pasted all available permission constants in this package and the only one that worked was "Carbohydrates".

Also **thank you for this package**! I am not very familiar with iOS development, but if Carbohydrates work, I think I can try helping fix this problem, given it's a bug and not missconfiguration by me...

--- a/RCTAppleHealthKit/RCTAppleHealthKit+TypesAndPermissions.m
+++ b/RCTAppleHealthKit/RCTAppleHealthKit+TypesAndPermissions.m
@@ -167,11 +167,6 @@
         return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierOxygenSaturation];
     }
 
-    // Results Identifiers
-    if ([@"BloodGlucose" isEqualToString: key]) {
-        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierBloodGlucose];
-    }
-
     // Sleep
     if ([@"SleepAnalysis" isEqualToString: key]) {
         return [HKObjectType categoryTypeForIdentifier:HKCategoryTypeIdentifierSleepAnalysis];

--- a/RCTAppleHealthKit/RCTAppleHealthKit+TypesAndPermissions.m
+++ b/RCTAppleHealthKit/RCTAppleHealthKit+TypesAndPermissions.m
@@ -64,10 +64,84 @@
     }
 
     // Nutrition Identifiers
-    if ([@"DietaryEnergy" isEqualToString: key]) {
-        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryEnergyConsumed];
-    } else if ([@"Carbohydrates" isEqualToString: key]) {
+    if ([@"Biotin" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryBiotin];
+    } else if ([@"Caffeine" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryCaffeine];
+    } else if ([@"Calcium" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryCalcium];
+    } else if ([@"Carbohydrates" isEqualToString:key]) {
         return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryCarbohydrates];
+    } else if ([@"Chloride" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryChloride];
+    } else if ([@"Cholesterol" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryCholesterol];
+    } else if ([@"Copper" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryCopper];
+    } else if ([@"EnergyConsumed" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryEnergyConsumed];
+    } else if ([@"FatMonounsaturated" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryFatMonounsaturated];
+    } else if ([@"FatPolyunsaturated" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryFatPolyunsaturated];
+    } else if ([@"FatSaturated" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryFatSaturated];
+    } else if ([@"FatTotal" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryFatTotal];
+    } else if ([@"Fiber" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryFiber];
+    } else if ([@"Folate" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryFolate];
+    } else if ([@"Iodine" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryIodine];
+    } else if ([@"Iron" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryIron];
+    } else if ([@"Magnesium" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryMagnesium];
+    } else if ([@"Manganese" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryManganese];
+    } else if ([@"Molybdenum" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryMolybdenum];
+    } else if ([@"Niacin" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryNiacin];
+    } else if ([@"PantothenicAcid" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryPantothenicAcid];
+    } else if ([@"Phosphorus" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryPhosphorus];
+    } else if ([@"Potassium" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryPotassium];
+    } else if ([@"Protein" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryProtein];
+    } else if ([@"Riboflavin" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryRiboflavin];
+    } else if ([@"Selenium" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietarySelenium];
+    } else if ([@"Sodium" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietarySodium];
+    } else if ([@"Sugar" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietarySugar];
+    } else if ([@"Thiamin" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryThiamin];
+    } else if ([@"VitaminA" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryVitaminA];
+    } else if ([@"VitaminB12" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryVitaminB12];
+    } else if ([@"VitaminB6" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryVitaminB6];
+    } else if ([@"VitaminC" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryVitaminC];
+    } else if ([@"VitaminD" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryVitaminD];
+    } else if ([@"VitaminE" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryVitaminE];
+    } else if ([@"VitaminK" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryVitaminK];
+    } else if ([@"Zinc" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryZinc];
+    } else if ([@"Water" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierDietaryWater];
+    } else if ([@"BloodGlucose" isEqualToString:key]) {
+        return [HKObjectType quantityTypeForIdentifier:HKQuantityTypeIdentifierBloodGlucose];
     }
 
     // Vital Signs Identifiers
@@ -248,7 +322,7 @@
         return [HKObjectType categoryTypeForIdentifier:HKCategoryTypeIdentifierMindfulSession];
     }
 
-		// Workout
+    // Workout
     if ([@"Workout" isEqualToString:key]) {
         return [HKObjectType workoutType];
     }

